### DISCUSSION TITLE
Enabling garbage collection in flux-system kustomization

### DISCUSSION THREAD

**Discussion by dmitriishaburov** [fluxcd/flux2#2079] [Q&A] [answered]
Hello!
We're using the `flux-system` kustomization as main source of all manifests for our cluster.
I'd like to enable garbage collection on it, and definetely can do it by adding the `prune` field to spec, but obviously it would be overwritten on next update, right?

What's the suggested/proper way to use garbage collection in the cluster?

**Comment by stefanprodan** [comment, marked as answer]
Here are the docs on how to customise the Flux manifests: https://fluxcd.io/docs/installation/#customize-flux-manifests

PS. GC is enabled by default when using `flux bootstrap`

**Reply by dmitriishaburov** [reply]
@stefanprodan Thanks for the reply, it's indeed works by default - just tested it with plain yaml manifests.
But I've deleted `Kustomization` that contains some resources and that resources stay in cluster. Is that a known issue?

**Reply by stefanprodan** [reply]
For any other Flux Kustomization besides `flux-system` presents in your repository, you have to set `prune: true`.

**Reply by dmitriishaburov** [reply]
Hm, I mean inside the folder of `flux-system` kustomization I have the subfolder with kustomization.yaml:
```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
  - clusterrole.yaml
  - clusterrolebinding.yaml
```
And when I delete this `kustomization.yaml`, `clusterrole.yaml` and `clusterrolebinding.yaml` they stay in cluster anyway.

**Reply by stefanprodan** [reply]
Is this overlay included in `flux-system/kustomization.yaml`?

**Reply by dmitriishaburov** [reply]
Nope, it's just stored in folder, and work properly in terms of applying and modifying resources, but not deleting.
Example usecase:
```
/live/flux-system/*

/live/grafana1/kustomization.yaml
/live/grafana1/patch.yaml

/live/grafana2/kustomization.yaml
/live/grafana2/patch.yaml
```

The grafana1/grafana2 kustomizations are not referenced anywhere

**Reply by stefanprodan** [reply]
Please see here how to structure your repo: https://github.com/fluxcd/flux2-kustomize-helm-example

No idea how those manifests got on the cluster, besides the Flux manifests (gotk-components and gotk-sync) no other thing should be reconciled by the root Kustomization.

**Reply by dmitriishaburov** [reply]
We have `  path: ./kubernetes/live` in `./kubernetes/live/flux-system/gotk-sync.yaml` file, and it applies all manifests (and kustimizations) that are in `kubernetes/live`, I though it is designed way to use it (as I've used flux v1) 😄

**Reply by stefanprodan** [reply]
I don't see how Flux would apply things inside `flux-system` sub-dirs if those overlays are not included in the `kustomization.yaml`...

**Reply by dmitriishaburov** [reply]
It is not inside the `flux-system`, if I running the `flux bootstrap github --path kubernetes/live`, it creates the `kubernetes/live/flux-system` folder, but actually applies everything that's inside the `kubernetes/live`, not only in `kubernetes/live/flux-system`
